rest/src/test/java/com/carhouse/rest/controllerIT/CarControllerTestIT.java

```diff
@@ -0,0 +1,125 @@
+package com.carhouse.rest.controllerIT;
+
+import com.carhouse.model.*;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.sql.Date;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CarControllerTestIT {
+
+    private static final String HOST = "http://localhost:8080";
+    private static final String CAR_LIST_GET_URL = "/carSale/car";
+    private static final String CAR_GET_URL = "/carSale/car/";
+    private static final String CAR_ADD_URL = "/carSale/car";
+    private static final String CAR_UPDATE_URL = "/carSale/car";
+    private static final String CAR_DELETE_URL = "/carSale/car/";
+
+    RestTemplate restTemplate = new RestTemplate();
+
+    @Test
+    void getCars() {
+        ResponseEntity<String> response = restTemplate.getForEntity(HOST + CAR_LIST_GET_URL, String.class);
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+    @Test
+    void getCar() {
+        ResponseEntity<Car> response = restTemplate.getForEntity(HOST + CAR_GET_URL + 1, Car.class);
+        assertEquals(200, response.getStatusCodeValue());
+        assertNotNull(response.getBody());
+    }
+
+    @Test
+    void getNotExistCarSale() {
+        assertThrows(HttpClientErrorException.NotFound.class, () -> restTemplate.getForEntity(HOST
+                + CAR_GET_URL + 150, Car.class));
+    }
+
+    @Test
+    void addCar() {
+        Car car = new Car(2);
+        car.setYear(Date.valueOf("2017-04-05"));
+        car.setCarModel(new CarModel(2, new CarMake(2)));
+        car.setFuelType(new FuelType(1));
+        car.setTransmission(new Transmission(2));
+        HttpEntity<Car> request = new HttpEntity<>(car);
+        ResponseEntity<String> response = restTemplate.postForEntity(HOST + CAR_ADD_URL, request, String.class);
+        assertEquals(200, response.getStatusCodeValue());
+        Integer id = Integer.valueOf(response.getBody());
+        assertTrue(id > 0);
+        restTemplate.delete(HOST + CAR_DELETE_URL + id);
+    }
+
+    @Test
+    void addCarWithWrongReference() {
+        Car car = new Car(2);
+        car.setYear(Date.valueOf("2017-04-05"));
+        car.setCarModel(new CarModel(20, new CarMake(12)));
+        car.setFuelType(new FuelType(15));
+        car.setTransmission(new Transmission(32));
+        HttpEntity<Car> request = new HttpEntity<>(car);
+        assertThrows(HttpClientErrorException.class, () -> restTemplate.postForEntity(HOST
```

Comment:
it would be good to check what kind of error has happened here.

Reply:
I've added these checks